src/dhcp_fsm.erl

```diff
@@ -316,7 +326,10 @@ delegate(F, Pkg, State) ->
 delegate(F, Pkg, Opts, State = #state{handler = M}) ->
     RPkg = lists:foldl(fun(O, P) ->
                                dhcp_package:set_option(O, P)
-                       end, dhcp_package:clone(Pkg), [{dhcp_server_identifier, State#state.server_identifier} | Opts]),
+                       end,
+                       dhcp_package:clone(Pkg),
+                       [{dhcp_server_identifier, State#state.server_identifier}
+                        | Opts]),
     RPkg1 = dhcp_package:set_op(reply, RPkg),
     case  M:F(RPkg1, Pkg, State#state.handler_state) of
```

Comment:
Remove the dynamic function call on line 334. Only modules that define callbacks should make dynamic calls.